<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1047</width>
    <height>802</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: #1b1d23</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="DesignPage">
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_17" stretch="1,1">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_10" stretch="0,1,0,0,0,0,0">
            <property name="leftMargin">
             <number>20</number>
            </property>
            <property name="rightMargin">
             <number>20</number>
            </property>
            <item>
             <widget class="QLabel" name="label_19">
              <property name="font">
               <font>
                <family>MS Shell Dlg 2</family>
                <pointsize>11</pointsize>
               </font>
              </property>
              <property name="styleSheet">
               <string notr="true">color:white;
font-weight:10000px;</string>
              </property>
              <property name="text">
               <string>Z_Plane Circle </string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="PlotWidget" name="zPlaneCircle" native="true">
              <property name="styleSheet">
               <string notr="true">background-color: rgb(0, 0, 0);
</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_18">
              <item>
               <spacer name="horizontalSpacer_9">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QCheckBox" name="checkBox_4">
                <property name="font">
                 <font>
                  <family>MS Shell Dlg 2</family>
                  <pointsize>11</pointsize>
                 </font>
                </property>
                <property name="styleSheet">
                 <string notr="true">color:white;
font-weight:10000px;</string>
                </property>
                <property name="text">
                 <string>conjugates</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_10">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer_7">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>80</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_19">
              <item>
               <widget class="QRadioButton" name="radioButton_2">
                <property name="font">
                 <font>
                  <pointsize>11</pointsize>
                 </font>
                </property>
                <property name="styleSheet">
                 <string notr="true">color:white;</string>
                </property>
                <property name="text">
                 <string>Zeros</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="radioButton">
                <property name="font">
                 <font>
                  <pointsize>11</pointsize>
                 </font>
                </property>
                <property name="styleSheet">
                 <string notr="true">color:white;</string>
                </property>
                <property name="text">
                 <string>Poles</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_20">
              <item>
               <widget class="QComboBox" name="comboBox_2">
                <property name="styleSheet">
                 <string notr="true">QComboBox {
    font-family: &quot;Overpass&quot;;
    font-weight: 600; /* Semibold */
    font-size: 10pt; /* Adjust the font size as needed */
    color: white; /* Set the text color to white */
    border: 1px solid white; /* Set the border to 1px solid white */
background: rgba(74, 74, 74, 0);
	border-radius: 10px;
}</string>
                </property>
                <item>
                 <property name="text">
                  <string>Zeros</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Poles</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Both</string>
                 </property>
                </item>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="clear_4">
                <property name="font">
                 <font>
                  <family>-apple-system</family>
                  <pointsize>11</pointsize>
                 </font>
                </property>
                <property name="styleSheet">
                 <string notr="true">QPushButton{
  appearance: button;
  border-radius: 6px;
  border-width: 0;
  box-shadow: rgba(50, 50, 93, .1) 0 0 0 1px inset,rgba(50, 50, 93, .1) 0 2px 5px 0,rgba(0, 0, 0, .07) 0 1px 1px 0;
color: rgb(255, 255, 255);
background-color:rgb(22, 25, 76);
  cursor: pointer;
  font-family: -apple-system,system-ui,&quot;Segoe UI&quot;,Roboto,&quot;Helvetica Neue&quot;,Ubuntu,sans-serif;
  height: 30px;
  margin: 0 20px 0 20px;
  position: center;
  width:70px;
}
QPushButton:hover{
background-color: #283999;
}
QPushButton:pressed{
background-color: #1c2973;
}</string>
                </property>
                <property name="text">
                 <string>Clear</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer_8">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>80</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_11" stretch="0,4,0,4,0">
            <property name="leftMargin">
             <number>20</number>
            </property>
            <property name="rightMargin">
             <number>20</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="label_20">
              <property name="font">
               <font>
                <family>MS Shell Dlg 2</family>
                <pointsize>11</pointsize>
               </font>
              </property>
              <property name="styleSheet">
               <string notr="true">color:white;
font-weight:10000px;</string>
              </property>
              <property name="text">
               <string>Magnitude Response</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="PlotWidget" name="magPlot" native="true">
              <property name="styleSheet">
               <string notr="true">background-color: rgb(0, 0, 0);</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_21">
              <property name="font">
               <font>
                <family>MS Shell Dlg 2</family>
                <pointsize>11</pointsize>
               </font>
              </property>
              <property name="styleSheet">
               <string notr="true">color:white;
font-weight:10000px;</string>
              </property>
              <property name="text">
               <string>Phase Response</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="PlotWidget" name="phasePlot" native="true">
              <property name="styleSheet">
               <string notr="true">background-color: rgb(0, 0, 0);
</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="correctPhase_4">
              <property name="font">
               <font>
                <family>-apple-system</family>
                <pointsize>11</pointsize>
               </font>
              </property>
              <property name="styleSheet">
               <string notr="true">QPushButton{
  appearance: button;
  border-radius: 6px;
  border-width: 0;
  box-shadow: rgba(50, 50, 93, .1) 0 0 0 1px inset,rgba(50, 50, 93, .1) 0 2px 5px 0,rgba(0, 0, 0, .07) 0 1px 1px 0;
color: rgb(255, 255, 255);
background-color:rgb(22, 25, 76);
  cursor: pointer;
  font-family: -apple-system,system-ui,&quot;Segoe UI&quot;,Roboto,&quot;Helvetica Neue&quot;,Ubuntu,sans-serif;
  height: 30px;
  margin: 0 20px 0 20px;
  position: center;
  width:70px;
}
QPushButton:hover{
background-color: #283999;
}
QPushButton:pressed{
background-color: #1c2973;
}</string>
              </property>
              <property name="text">
               <string>Correct Phase</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="CorrectPage">
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="0" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_21" stretch="1,1">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_12" stretch="0,1,0,0,0,0,0,0,0,0">
            <property name="leftMargin">
             <number>20</number>
            </property>
            <property name="rightMargin">
             <number>20</number>
            </property>
            <item>
             <widget class="QLabel" name="label_24">
              <property name="font">
               <font>
                <pointsize>11</pointsize>
               </font>
              </property>
              <property name="styleSheet">
               <string notr="true">color:white;
font-weight:10000px;</string>
              </property>
              <property name="text">
               <string>Z_Plane Circle </string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="PlotWidget" name="AllpassZplane" native="true">
              <property name="styleSheet">
               <string notr="true">background-color: rgb(0, 0, 0);
</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_22"/>
            </item>
            <item>
             <spacer name="verticalSpacer_9">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>80</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QLabel" name="label">
              <property name="font">
               <font>
                <pointsize>11</pointsize>
               </font>
              </property>
              <property name="styleSheet">
               <string notr="true">color:white;
</string>
              </property>
              <property name="text">
               <string>All Pass Filters</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="InsertFilterText">
              <property name="styleSheet">
               <string notr="true">color:white;
</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_23"/>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_24">
              <item>
               <widget class="QComboBox" name="comboBox"/>
              </item>
              <item>
               <widget class="QPushButton" name="AddFilter">
                <property name="styleSheet">
                 <string notr="true">QPushButton{
  appearance: button;
  border-radius: 6px;
  border-width: 0;
  box-shadow: rgba(50, 50, 93, .1) 0 0 0 1px inset,rgba(50, 50, 93, .1) 0 2px 5px 0,rgba(0, 0, 0, .07) 0 1px 1px 0;
color: rgb(255, 255, 255);
background-color:rgb(22, 25, 76);
  cursor: pointer;
  font-family: -apple-system,system-ui,&quot;Segoe UI&quot;,Roboto,&quot;Helvetica Neue&quot;,Ubuntu,sans-serif;
  height: 30px;
  margin: 0 20px 0 20px;
  position: center;
  width:70px;
}
QPushButton:hover{
background-color: #283999;
}
QPushButton:pressed{
background-color: #1c2973;
}</string>
                </property>
                <property name="text">
                 <string>Add</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="Apply">
                <property name="styleSheet">
                 <string notr="true">QPushButton{
  appearance: button;
  border-radius: 6px;
  border-width: 0;
  box-shadow: rgba(50, 50, 93, .1) 0 0 0 1px inset,rgba(50, 50, 93, .1) 0 2px 5px 0,rgba(0, 0, 0, .07) 0 1px 1px 0;
color: rgb(255, 255, 255);
background-color:rgb(22, 25, 76);
  cursor: pointer;
  font-family: -apple-system,system-ui,&quot;Segoe UI&quot;,Roboto,&quot;Helvetica Neue&quot;,Ubuntu,sans-serif;
  height: 30px;
  margin: 0 20px 0 20px;
  position: center;
  width:70px;
}
QPushButton:hover{
background-color: #283999;
}
QPushButton:pressed{
background-color: #1c2973;
}</string>
                </property>
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="remove">
              <property name="styleSheet">
               <string notr="true">QPushButton{
  appearance: button;
  border-radius: 6px;
  border-width: 0;
  box-shadow: rgba(50, 50, 93, .1) 0 0 0 1px inset,rgba(50, 50, 93, .1) 0 2px 5px 0,rgba(0, 0, 0, .07) 0 1px 1px 0;
color: rgb(255, 255, 255);
background-color:rgb(22, 25, 76);
  cursor: pointer;
  font-family: -apple-system,system-ui,&quot;Segoe UI&quot;,Roboto,&quot;Helvetica Neue&quot;,Ubuntu,sans-serif;
  height: 30px;
  margin: 0 20px 0 20px;
  position: center;
  width:70px;
}
QPushButton:hover{
background-color: #283999;
}
QPushButton:pressed{
background-color: #1c2973;
}</string>
              </property>
              <property name="text">
               <string>Remove</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_10">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>80</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_13" stretch="0,4,0,4">
            <property name="leftMargin">
             <number>20</number>
            </property>
            <property name="rightMargin">
             <number>20</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="label_25">
              <property name="font">
               <font>
                <pointsize>11</pointsize>
               </font>
              </property>
              <property name="styleSheet">
               <string notr="true">color:white;
font-weight:10000px;</string>
              </property>
              <property name="text">
               <string>Corrected Phase</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="PlotWidget" name="Correctedphaseplot" native="true">
              <property name="styleSheet">
               <string notr="true">background-color: rgb(0, 0, 0);</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_26">
              <property name="font">
               <font>
                <pointsize>11</pointsize>
               </font>
              </property>
              <property name="styleSheet">
               <string notr="true">color:white;
font-weight:10000px;</string>
              </property>
              <property name="text">
               <string>Phase Response</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="PlotWidget" name="AllPassPhasePlot" native="true">
              <property name="styleSheet">
               <string notr="true">background-color: rgb(0, 0, 0);
</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="ApplyPage">
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="0" column="0">
         <widget class="QFrame" name="frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="0" column="0">
            <widget class="QFrame" name="frame_2">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QGridLayout" name="gridLayout_4">
              <item row="0" column="0">
               <widget class="QLabel" name="label_23">
                <property name="font">
                 <font>
                  <pointsize>14</pointsize>
                 </font>
                </property>
                <property name="styleSheet">
                 <string notr="true">color:white;</string>
                </property>
                <property name="text">
                 <string>Input</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="PlotWidget" name="InputSignal" native="true">
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>200</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true">background-color: rgb(0, 0, 0);
</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_22">
                <property name="font">
                 <font>
                  <pointsize>14</pointsize>
                 </font>
                </property>
                <property name="styleSheet">
                 <string notr="true">color:white;</string>
                </property>
                <property name="text">
                 <string>Output</string>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="PlotWidget" name="OutputFilteredSignal" native="true">
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>200</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true">background-color: rgb(0, 0, 0);
</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QFrame" name="frame_3">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QGridLayout" name="gridLayout_3">
              <item row="0" column="1">
               <widget class="QSlider" name="horizontalSlider">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item row="1" column="0" colspan="2">
               <widget class="QRadioButton" name="Touchpadcheckbox">
                <property name="font">
                 <font>
                  <pointsize>14</pointsize>
                 </font>
                </property>
                <property name="styleSheet">
                 <string notr="true">color:white;
font-weight:10000px;</string>
                </property>
                <property name="text">
                 <string>Touch Pad</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="label_2">
                <property name="font">
                 <font>
                  <pointsize>11</pointsize>
                 </font>
                </property>
                <property name="styleSheet">
                 <string notr="true">color:white;
</string>
                </property>
                <property name="text">
                 <string>FIlterSpeed</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0" colspan="2">
               <widget class="QGraphicsView" name="Touchpad">
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>300</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true">background-color: rgb(0, 0, 0);</string>
                </property>
               </widget>
              </item>
             </layout>
             <zorder>label_2</zorder>
             <zorder>horizontalSlider</zorder>
             <zorder>Touchpad</zorder>
             <zorder>Touchpadcheckbox</zorder>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1047</width>
     <height>26</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">color:white;</string>
   </property>
   <widget class="QMenu" name="menuDesign">
    <property name="styleSheet">
     <string notr="true">color:white;
</string>
    </property>
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
   </widget>
   <widget class="QMenu" name="menuImplement">
    <property name="styleSheet">
     <string notr="true">color:white;</string>
    </property>
    <property name="title">
     <string>Tab</string>
    </property>
    <addaction name="actionAdjust_Phase"/>
    <addaction name="actionDesign"/>
    <addaction name="actionApply"/>
   </widget>
   <widget class="QMenu" name="menuApply">
    <property name="title">
     <string>Apply</string>
    </property>
   </widget>
   <addaction name="menuDesign"/>
   <addaction name="menuImplement"/>
   <addaction name="menuApply"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Load</string>
   </property>
  </action>
  <action name="actionDesign">
   <property name="text">
    <string>Design</string>
   </property>
  </action>
  <action name="actionAdjust_Phase">
   <property name="text">
    <string>Correct Phase</string>
   </property>
  </action>
  <action name="actionApply">
   <property name="text">
    <string>Apply</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QWidget</extends>
   <header location="global">pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
